Moved non-translatable elements out of strings.json for nuki

sonianuj287 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.
- [x] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

sonianuj287 committed:
fix-translatable-urls-nuki

sonianuj287 commented on homeassistant/components/nuki/strings.json:
```diff
@@ -34,7 +34,7 @@
   "issues": {
```
It should only be config section in strings.json?

epenet reviewed (changes requested):
Update status accordingly

NoRi2909 commented on homeassistant/components/nuki/strings.json:
```diff
@@ -34,7 +34,7 @@
     "https_webhook": {
       "title": "Nuki webhook URL uses HTTPS (SSL)",
-      "description": "The Nuki bridge can not push events to an HTTPS address (SSL), please configure a (local) HTTP address under \"Home Assistant URL\" in the [network settings]({network_link}). The current (local) address is: `{base_url}`, a valid address could, for example, be `http://192.168.1.10:8123` where `192.168.1.10` is the IP of the Home Assistant device"
+      "description": "The Nuki bridge can not push events to an HTTPS address (SSL), please configure a (local) HTTP address under \"Home Assistant URL\" in the [network settings]({network_link}). The current (local) address is: `{base_url}`, a valid address could, for example, be {sample_ip} where `192.168.1.10` is the IP of the Home Assistant device"
```
As mentioned by me and @epenet in the previous PR and above you should use two placeholders here for `{sample_url}` and `{sample_ip}`. 

Replacing just one in the translation strings will not prevent inconsistencies in other languages.

sonianuj287 committed:
fixed config ips to config flow

sonianuj287 committed:
resolved copilot comments

epenet commented:
Also, please consider replacing the urls on the descriptions for the issues.

You just need to make sure that you update the translation placeholders in the right location:

https://github.com/home-assistant/core/blob/7bc43039bd2d05a1dfa128e68003bb55a2ade40b/homeassistant/components/nuki/__init__.py#L103

and

https://github.com/home-assistant/core/blob/7bc43039bd2d05a1dfa128e68003bb55a2ade40b/homeassistant/components/reolink/host.py#L433

sonianuj287 commented on homeassistant/components/nuki/config_flow.py:
```diff
@@ -140,7 +144,10 @@ async def async_step_reauth_confirm(
 
         return self.async_show_form(
-            step_id="reauth_confirm", data_schema=REAUTH_SCHEMA, errors=errors
+            step_id="reauth_confirm",
```
I'm raising PR for that in a minute

epenet replied:
Please revert the formatting herd to reduce the PR

sonianuj287 committed:
resolved comment

sonianuj287 committed:
replacing placeholder for issues

sonianuj287 committed:
fixed ci error

pantherale0 commented:
Also, you should split the reolink changes into a separate PR

sonianuj287 committed:
remove reolink changes

epenet commented on homeassistant/components/nuki/__init__.py:
```diff
@@ -110,6 +110,7 @@ async def handle_webhook(
             translation_placeholders={
                 "base_url": hass_url,
                 "network_link": "https://my.home-assistant.io/redirect/network/",
+                "sample_ip": "http://192.168.1.10:8123",
```
This is not an IP. It's a URL.
It should instead be called `sample_url`.

You will need to provide both a sample IP and a sample URL for the issue.

sonianuj287 committed:
Fix sample_ip vs sample_url distinction in Nuki issue descriptions

- Rename placeholder from sample_ip to sample_url for the full URL (http://192.168.1.10:8123)
- Add separate sample_ip placeholder for just the IP address (192.168.1.10)
- Update issue description to use sample_url for the example URL and sample_ip for the IP reference
- Addresses reviewer feedback about incorrect naming of URL as IP

sonianuj287 committed:
Merge branch 'dev' into fix-translatable-urls-nuki

joostlek committed:
Update homeassistant/components/nuki/config_flow.py

joostlek commented on homeassistant/components/nuki/config_flow.py:
```diff
@@ -140,7 +142,9 @@ async def async_step_reauth_confirm(
-            step_id="reauth_confirm", data_schema=REAUTH_SCHEMA, errors=errors
+            step_id="reauth_confirm",
+            data_schema=REAUTH_SCHEMA,
+            errors=errors,
```
```suggestion
            step_id="reauth_confirm", data_schema=REAUTH_SCHEMA, errors=errors
```

joostlek committed:
Update homeassistant/components/nuki/config_flow.py

epenet committed:
Update homeassistant/components/nuki/strings.json

NoRi2909 commented on homeassistant/components/nuki/strings.json:
```diff
@@ -34,7 +34,7 @@
     "https_webhook": {
       "title": "Nuki webhook URL uses HTTPS (SSL)",
-      "description": "The Nuki bridge can not push events to an HTTPS address (SSL), please configure a (local) HTTP address under \"Home Assistant URL\" in the [network settings]({network_link}). The current (local) address is: `{base_url}`, a valid address could, for example, be `http://192.168.1.10:8123` where `192.168.1.10` is the IP of the Home Assistant device"
+      "description": "The Nuki bridge can not push events to an HTTPS address (SSL), please configure a (local) HTTP address under \"Home Assistant URL\" in the [network settings]({network_link}). The current (local) address is: `{base_url}`, a valid address could, for example, be `{sample_url}` where `{sample_ip}` is the IP of the Home Assistant device"
```
Here "can not" should be replaced with "cannot" as it's definitive here:

```suggestion
      "description": "The Nuki bridge cannot push events to an HTTPS address (SSL), please configure a (local) HTTP address under \"Home Assistant URL\" in the [network settings]({network_link}). The current (local) address is: `{base_url}`, a valid address could, for example, be `{sample_url}` where `{sample_ip}` is the IP of the Home Assistant device"
```

So we can fix this along the way.

sonianuj287 committed:
Update homeassistant/components/nuki/strings.json

Co-authored-by: Norbert Rittel <norbert@rittel.de>

sonianuj287 committed:
Merge branch 'dev' into fix-translatable-urls-nuki

sonianuj287 committed:
Merge branch 'dev' into fix-translatable-urls-nuki

sonianuj287 committed:
Merge branch 'dev' into fix-translatable-urls-nuki

sonianuj287 committed:
Merge branch 'dev' into fix-translatable-urls-nuki

sonianuj287 commented:
Hi @epenet @NoRi2909 @joostlek , could you please verify the PR and move this to merge. 
Thanks :)

epenet commented on homeassistant/components/nuki/config_flow.py:
```diff
@@ -19,6 +19,8 @@
 _LOGGER = logging.getLogger(__name__)
 
+SAMPLE_IP = "192.168.1.25"
+
```
```suggestion
```

epenet committed:
Apply suggestions from code review
